packages/utils/test-tools/src/enzymeTests.ts

```diff
@@ -0,0 +1,71 @@
+import { mount, ReactWrapper } from 'enzyme';
+
+export type JSXProducer = () => JSX.Element;
+
+export interface PropTreeFilter {
+  children?: boolean;
+  functions?: boolean;
+}
+
+export type PropTreeSnapshot = {
+  name: string;
+  props: object;
+  children: PropTreeSnapshot[];
+};
+
+export function snapshotPropTree(node: ReactWrapper, recurse: number, filter: PropTreeFilter = {}): PropTreeSnapshot {
+  // get a potentially filtered copy of the props
+  const nodeProps = node.props();
+  const props = {};
+  Object.keys(nodeProps).forEach((key) => {
+    if (!(filter.children && key === 'children') && !(filter.functions && typeof nodeProps[key] === 'function')) {
+      props[key] = nodeProps[key];
+    }
+  });
+
+  // then build the base result
+  const result: PropTreeSnapshot = { name: node.name(), props, children: [] };
+
+  // if children are requested add children
+  if (recurse > 0) {
+    node.children().forEach((child) => {
+      result.children.push(snapshotPropTree(child, recurse - 1, filter));
+    });
+  }
+  return result;
+}
+
+export function compareTrees(a: PropTreeSnapshot, b: PropTreeSnapshot, paths: string[]): void {
+  const newPaths = paths.concat(a.name);
+  if (a.name !== b.name) {
+    throw new Error(`Shallow compare found two nodes with different names at ${paths.join(': ')}`);
+  }
+  if (a.children.length !== b.children.length) {
+    throw new Error(`Shallow compare found two nodes at ${paths.join(': ')} with different property counts`);
+  }
+  Object.keys(a.props).forEach((key) => {
```

Comment:
for full correctness, don't you also need to check that all of `b.props` are covered by `a.props`?

Reply:
Yep, good catch. I had the check in a previous iteration but lost it when I added the tree snapshotting. I will add a test for Object.keys(a.props).length === Object.keys(b.props).length to ensure things are covered